<launch>
<arg name="robot" default="$(env ROBOT)"/>
<arg name="EKF_on" default="1"/>
<arg name="hold_time" default="8.0"/>
<arg name="resolution" default="1"/>
<arg name="turn_speed" default="0.3"/>
<arg name="send" default="1"/>
<arg name="drive_speed" default="0.2"/>
<arg name="angle_tolerance" default="0.5"/>
<arg name="radius" default="50"/>
<arg name="distance_tolerance" default="0.10"/>
<arg name="show_cells_" default="false"/>
<group ns="$(arg robot)">
<param name="tf_prefix" value="$(arg robot)"/>
<!-- Minimal -->
<include file="$(find turtlebot_bringup)/launch/minimal.launch">
</include>
<param name="mobile_base/base_frame" value="$(arg robot)/base_footprint"/>
<param name="mobile_base/odom_frame" value="$(arg robot)/odom"/>

<!-- Communication EKF node -->
<node pkg="turtlebot_deployment" type="inter_robot_communication" name="communication">
<param name="robot_name" value="$(arg robot)"/>
<param name="EKF_switch" value="$(arg EKF_on)"/>

</node>
<!-- EKF -->
<node pkg="turtlebot_deployment" type="ekf" name="ekf_filter" output="screen">
<param name="robot_name"	value="$(arg robot)"/>
<param name="sendAll"	value="$(arg send)"/>
</node>

<!--Cyclic Pursuit-->
<node pkg="turtlebot_deployment" type="cyclic_pursuit" name="cyclic_pursuit">
<param name="robot_name" value="$(arg robot)"/>
<param name="hold_time" value="$(arg hold_time)"/>
<param name="resolution" value="$(arg resolution)"/>
<param name="show_cells_" value="$(arg show_cells_)"/>
</node>
<!--PathFollowing-->
<node pkg="turtlebot_deployment" type="PathFollowing" name="PathFollowing">
<param name="robot_name" value="$(arg robot)"/>
<param name="Radius" value="$(arg radius)"/>
<param name="resolution" value="$(arg resolution)"/>
<param name="show_cells_" value="$(arg show_cells_)"/>
</node>
</group>
</launch>
